test/uk/gov/hmrc/gatekeeperemail/controllers/GatekeeperComposeEmailControllerSpec.scala

```diff
@@ -131,13 +131,15 @@ class GatekeeperComposeEmailControllerSpec extends AnyWordSpec with Matchers wit
       when(mockEmailConnector.sendEmail(*)).thenReturn(successful(Status.OK))
       when(mockEmailRepository.persist(*)).thenReturn(Future(InsertOneResult.acknowledged(BsonNumber(1))))
       when(mockEmailRepository.updateEmailSentStatus(*)).thenReturn(successful(email))
+      when(mockSentEmailRepository.persist(*)).thenReturn(Future(InsertOneResult.acknowledged(BsonNumber(1))))
       val result = controller.sendEmail(emailUUID)(fakeRequest)
       status(result) shouldBe Status.OK
     }
 
     "return 500" in new Setup {
-      when(mockEmailConnector.sendEmail(*)).thenReturn(failed(new IOException("can not connect to email service")))
-      when(mockEmailRepository.persist(*)).thenReturn(Future(InsertOneResult.acknowledged(BsonNumber(1))))
+      when(mockEmailRepository.getEmailData(*)).thenReturn(failed(new IOException("can not connect to email service")))
```

Comment:
This message seems confusing as the emailRepository doesn't connect to the email service.

Reply:
Come to think of it, why aren't we just mocking the EmailService, instead of relying on mocking its collaborators?